### ISSUE TITLE
[Accessibility][ContextualMenu] Items loose focus when items are updated with out dismissing the menu

### ISSUE THREAD

**Issue by yaananth** [OfficeDev/office-ui-fabric-react#1324]
### Bug Report

- __Package version(s)__: latest

#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? (Yes, No) Yes

Requested priority: (Blocking, High, Normal, Low) Low

#### Describe the issue:

http://codepen.io/yashanantha/pen/VpxwRO

It's just a little hack demo to show the issue, go the demo, click anywhere in the "white area" (where results are displayed), press `tab` and press `enter`.
You can see that the first item is focused by default, good.
Now use arrow to move to the second item.
Press `enter`. (This essentially keeps the menu mounted, but just updates the items).
Focus is lost!

#### Actual behavior:

Focus is lost when items are updated. (keeping the contextmenu still mounted)

#### Expected behavior:

Every item the component is updated, focus should be set to the first child.

Looks like `shouldFocusOnMount` eventually gets to `CalloutContent` component to as `setInitialFocus` prop.

Where it has
```
public componentDidUpdate() {
    this._setInitialFocus();
    this._updatePosition();
  }
```
But

```
protected _setInitialFocus() {
    if (this.props.setInitialFocus && !this._didSetInitialFocus && this.state.positions) {
      this._didSetInitialFocus = true;
      focusFirstChild(this._calloutElement);
    }
  }
```
ensures that we try to set focus only **once** initially.
Why is that? Do we need that check at all? I think trying to focus every time when the component is updated **(if** there is no existing focus ofcourse) makes sense to me.

**Comment by MLoughry** [comment]
This happens because the item that is focused (determined by its `key` value) is removed. If an item is changed (but not removed), focus remains.

The problem with setting focus other than on initial mounting is that a rerender can  steal focus away from someplace else.
